crl/crl_x509/crl_test.go

```diff
@@ -352,9 +352,41 @@ func TestCreateRevocationList(t *testing.T) {
 					tc.template.SignatureAlgorithm)
 			}
 
-			if !reflect.DeepEqual(parsedCRL.RevokedCertificates, tc.template.RevokedCertificates) {
-				t.Fatalf("RevokedCertificates mismatch: got %v; want %v.",
-					parsedCRL.RevokedCertificates, tc.template.RevokedCertificates)
+			if len(parsedCRL.RevokedCertificates) != len(tc.template.RevokedCertificates) {
+				t.Fatalf("RevokedCertificates length mismatch: got %d; want %d.",
+					len(parsedCRL.RevokedCertificates), len(tc.template.RevokedCertificates))
+			}
+			for i, rc := range parsedCRL.RevokedCertificates {
+				erc := tc.template.RevokedCertificates[i]
+				if rc.SerialNumber.String() != erc.SerialNumber.String() {
+					t.Fatalf("RevokedCertificates entry %d serial mismatch: got %s; want %s.",
```

Comment:
This and below should be Errorf. That allows collecting multiple errors per test run, vs Fatal which exits immediately.

Note that will cause a little interaction with the numExtra logic, which will need a tweak.

Reply:
Ah, I was wrong about the numExtra. I was thinking it depended on the two `if` statements above it `Fatal`ing out; but actually it's correct in all cases since it's just saying "if there's a non-nil reasonCode, that's an extension not listed in ExtraExtensions, so we should count it." 👍🏻

I was thinking the lines below, like 366, 372, etc. should also change from Fatalf to Errorf.